assume/scenario/loader_pypsa.py

```diff
@@ -0,0 +1,232 @@
+# SPDX-FileCopyrightText: ASSUME Developers
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import shutil
+from datetime import datetime, timedelta
+from pathlib import Path
+
+import pandas as pd
+import pypsa
+from dateutil import rrule as rr
+
+from assume import World
+from assume.common.forecasts import NaiveForecast
+from assume.common.market_objects import MarketConfig, MarketProduct
+
+
+async def load_pypsa_async(
+    world: World,
+    scenario: str,
+    study_case: str,
+    network: pypsa.Network,
```

Comment:
why is network passed? should we use the same structure as other loaders, where we specify the path, and network and other things are loaded from there. I don't find this approach in line with the rest of the framework

Reply:
The idead is to have a network loaded from somewhere.
Whatever is available here:
https://pypsa.readthedocs.io/en/latest/import_export.html
* HDF5
* PyPower
* Pandapower
* netCDF

If you want to load a pypsa Network from CSV you can use this:
https://pypsa.readthedocs.io/en/latest/api_reference.html#pypsa.Network.import_from_csv_folder
and put the loaded network into the `pypsa_loader`